src/jit/decomposelongs.cpp

```diff
@@ -796,11 +799,9 @@ GenTree* DecomposeLongs::DecomposeInd(LIR::Use& use)
         addrBase->TypeGet(), addrBase->AsLclVarCommon()->GetLclNum(), BAD_IL_OFFSET);
     GenTreePtr addrHigh = new(m_compiler, GT_LEA) GenTreeAddrMode(TYP_REF, addrBaseHigh, nullptr, 0, genTypeSize(TYP_INT));
     GenTreePtr indHigh = new (m_compiler, GT_IND) GenTreeIndir(GT_IND, TYP_INT, addrHigh, nullptr);
-    
+
     // Insert the nodes into the block
-    m_blockRange.InsertAfter(addrBaseHigh, indLow);
-    m_blockRange.InsertAfter(addrHigh, addrBaseHigh);
-    m_blockRange.InsertAfter(indHigh, addrHigh);
+    m_blockRange.InsertAfter(LIR::SeqTree(m_compiler, indHigh), indLow);
```

Comment:
> In the StoreInd case, the operand is removed from the range before the new tree is sequenced, then re-inserted by the InsertAfter code ("code motion" of the local/leaf),

Hm. That code motion is not necessarily safe, and needs to be checked if it's going to be performed. In any case, the `dataLow` operand is not removed from the tree. I may have added this bug when I was working on decomp :(

> In the DecomposeInd case, there is also no sharing, as the "addrBaseHigh" creates a new GT_LCL_VAR reference.

Yup, I missed that. I agree that the `DecomposeInd` case is safe.